drivers/hmis/app/models/hmis/form/form_processor.rb

```diff
@@ -154,20 +161,34 @@ def current_living_situation_factory(create: true)
 
     return current_living_situation if current_living_situation.present? || !create
 
-    self.current_living_situation = enrollment_factory.current_living_situations.build(user_id: custom_assessment&.user_id)
+    self.current_living_situation = enrollment_factory.current_living_situations.build
   end
 
   def service_factory(create: true) # rubocop:disable Lint/UnusedMethodArgument
-    @service_factory ||= owner.owner if owner.is_a? Hmis::Hud::HmisService
+    @service_factory ||= owner if owner.is_a?(Hmis::Hud::Service) || owner.is_a?(Hmis::Hud::CustomService)
   end
 
   # Type Factories
+
+  # Enrollment is a special case, because it is not referenced by the FormProcessor.
+  # Enrollment can be the owner, or it can be related to the owner.
+  # Examples of forms that may update Enrollment:
+  #  - set EntryDate from intake assessment (owner_type=CustomAssessment)
+  #  - enroll an existing client (owner_type=Enrollment)
+  #  - update Move-in Date at occurrence (owner_type=Enrollment)
+  #
+  # In some other cases, the enrollment_factory can be used indirectly, to determine the relationship between records.
+  # An example is a CustomCaseNote form that generates a CurrentLivingSituation. The form does not update Enrollment directly,
+  # but it relies on the enrollment_factory to determine which enrollment to use when generating a new CLS (current_living_situation_factory)
```

Comment:
This is weird so figured I'd explain it. Maybe I could come up with something cleaner. I hesitated to dive to far into "storing related records for non-assessment form submissions" since that's NOT what this ticket was originally about, but I did add one test for it to submit_form_spec (which tests the CaseNote scenario described above). 🌀